flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -523,11 +534,15 @@ protected void processInput(MailboxDefaultAction.Controller controller) throws E
         if (!recordWriter.isAvailable()) {
             timer = new GaugePeriodTimer(ioMetrics.getBackPressuredTimePerSecond());
             resumeFuture = recordWriter.getAvailableFuture();
-        } else {
+        } else if (!inputProcessor.isAvailable()) {
             timer =
                     new ThroughputPeriodTimer(
                             ioMetrics.getIdleTimeMsPerSecond(), throughputCalculator);
             resumeFuture = inputProcessor.getAvailableFuture();
+        } else {
+            // todo: add new metrics (FLINK-23486)
+            timer = new GaugePeriodTimer(ioMetrics.getBackPressuredTimePerSecond());
+            resumeFuture = changelogWriterAvailabilityProvider.getAvailableFuture();
```

Comment:
For the sake of simplicity, in this PR and in the MVP maybe it would be best to account time when changelog/state backend is not available as `busy`? This would be easier to implement as it would naturally account consistently waiting on the availability future and on `lock.wait();`.